core/src/main/java/org/apache/brooklyn/policy/InvokeEffectorOnSensorChange.java

```diff
@@ -79,18 +80,29 @@ public void setEntity(EntityLocal entity) {
             LOG.debug("Defaulting to producer==self for {}, on entity {}", this, entity);
             producer = entity;
         }
+        lastValue = entity.sensors().get(sensor);
```

Comment:
This is where last known value is expected to be recovered after rebind.

Reply:
Good point - I missed this